src/bin/cql-stress-cassandra-stress/settings/param/parser.rs

```diff
@@ -0,0 +1,251 @@
+use anyhow::Result;
+use std::{cell::RefCell, rc::Rc};
+
+use super::{
+    simple_param::{SimpleParam, SimpleParamHandle},
+    ParamCell, ParamHandle, ParamParseResult,
+};
+
+/// Some of the parameters are mutually exclusive. For example, we can't do
+/// `./cassandra-stress read n=10 n<20`.
+/// That's why we arrange the parameters in so called groups.
+///
+/// Given the output of `./cassandra-stress help read`:
+///
+/// Usage: read [err<?] [n>?] [n<?] [no-warmup] [truncate=?] [cl=?] [serial-cl=?]
+///  OR
+/// Usage: read n=? [no-warmup] [truncate=?] [cl=?] [serial-cl=?]
+///  OR
+/// Usage: read duration=? [no-warmup] [truncate=?] [cl=?] [serial-cl=?]
+///
+/// We see that there are 3 groups of the parameters - a group can be identified on whether
+/// `n=` or `duration=` parameter has been defined.
+struct ParamsGroup {
+    params: Vec<ParamCell>,
+}
+
+impl ParamsGroup {
+    fn new(params: Vec<ParamCell>) -> Self {
+        Self { params }
+    }
+
+    fn parse(&mut self, arg: &str) -> ParamParseResult {
+        for param in self.params.iter() {
+            match param.borrow_mut().parse(arg) {
+                ParamParseResult::NoMatch => (),
+                v => return v,
+            }
+        }
+        ParamParseResult::NoMatch
+    }
+
+    /// Tells whether the group is satisfied with received arguments.
+    /// The group is satisfied \iff all of the following requirements are satisfied:
+    ///   - number of consumed parameters by the group is equal to the number of CLI arguments
+    ///   - every required parameter has been provided by the user
+    fn satisfied(&self, args_size: usize) -> bool {
+        let consumed_count = self
+            .params
+            .iter()
+            .filter(|p| p.borrow().supplied_by_user())
+            .count();
+        if consumed_count != args_size {
+            return false;
+        }
+        if self
+            .params
+            .iter()
+            .any(|param| param.borrow().required() && !param.borrow().supplied_by_user())
+        {
+            return false;
+        }
+
+        // The group is satisfied - it means that the parameters of this group
+        // were successfully parsed and will be returned to the user.
+        for param in self.params.iter() {
+            param.borrow_mut().set_parsed();
```

Comment:
`set_parsed()` explained here: https://github.com/scylladb/cql-stress/pull/40#discussion_r1267851172.

Regarding this comment:
I agree that this function does more than it should. I'll refactor the code so it will check whether the group is satisfied. Then, the parser would do something like 
```
if let Some(satisfied_group) = self.groups.iter().find(|g| g.satisfied(args_size)) {
    satisfied_group.mark_params_as_parsed(); // obviously these method will have some more descriptive name, just wanted to stick with the set_parsed() naming
    return Ok(());
}

Err(())
```

Sounds reasonable?

Reply:
Sounds better.